### PR TITLE
added link ref for stateless

### PR DESCRIPTION
**Pull request by viveksahu26**
Signed-off-by: Vivek Kumar Sahu <vivekkumarsahu650@gmail.com>

fixes https://github.com/knative/docs/issues/5403

Added link or reference for term or definition `Stateless` as mentioned in this [section](https://knative.dev/docs/getting-started/about-knative-functions/#about-knative-functions). The link should be provided for some of the definition which are pretty hard to understand. Specially it helps ones who are beginners. So providing link or ref for some particular definition helps in understanding the term properly.

## Proposed Changes 

- Added ref for the term or definition Stateless
- This is the Link: https://www.redhat.com/en/topics/cloud-native-apps/stateful-vs-stateless

### CODE REVIEW

File: docs/snippets/about-functions.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 -->
 Knative Functions provides a simple programming model for using functions on Knative, without requiring in-depth knowledge of Knative, Kubernetes, containers, or dockerfiles.
 
-Knative Functions enables you to easily create, build, and deploy stateless, event-driven functions as Knative Services by using the `func` CLI.
+Knative Functions enables you to easily create, build, and deploy [stateless](https://www.redhat.com/en/topics/cloud-native-apps/stateful-vs-stateless), event-driven functions as Knative Services by using the `func` CLI.
```
[the remark this answers is not in the document]
Comment (viveksahu26) [reply docs/snippets/about-functions.md:6]: Hmm, may be that's not the good idea to add other vendor links. Can we create separate docs place for docs like definition where we can add those definition by own.

[the remark this answers is not in the document]
Comment (psschwei) [reply docs/snippets/about-functions.md:6]: That's probably a discussion worth having with the docs working group leads @snneji @abrennan89

[the remark this answers is not in the document]
Comment (psschwei) [reply docs/snippets/about-functions.md:6]: Open an issue for it, and then link it on the [#knative-documentation slack channel](https://cloud-native.slack.com/archives/C04LY5G9ED7)

[the remark this answers is not in the document]
Comment (viveksahu26) [reply docs/snippets/about-functions.md:6]: https://github.com/knative/docs/issues/5403

[the remark this answers is not in the document]
Comment (evankanderson) [reply docs/snippets/about-functions.md:6]: I think both @snneji  and @abrennan89 are current busy, but I'd tend to agree that it's a better look to link to vendor-neutral definitions and documentation.

I'd probably prefer to build our own glossary -- the first link I found on Neeva was from 2019 and referenced a wikipedia article that doesn't seem to exist anymore.

[the remark this answers is not in the document]
Comment (abrennan89) [reply docs/snippets/about-functions.md:6]: Agreed re vendor neutral docs. The only place we really should be linking to vendor docs from is here: https://knative.dev/docs/install/knative-offerings/

RE having a glossary, this is a discussion that repeats a lot 🙂 I've even tried starting Twitter threads to see if we can gather _ANY_ insight into what terms folks would like to see defined, but to no avail. Outcome of the last discussion on this with the docs WG: https://github.com/knative/docs/issues/170#issuecomment-1169147031_

Doesn't mean we couldn't start one with this as the first definition I guess?

**Comment by abrennan89** [comment]
Closing this PR since we won't add links to vendor specific docs outside of the offerings page

**Pull request closed without merging**